workflow-templates/backport.yml

```diff
@@ -1,8 +1,15 @@
 name: Backport
 
 on:
-  pull_request:
+  # NOTE(negz): This is a risky target, but we run this action only when and if
+  # a PR is closed, then filter down to specifically merged PRs. We also don't
+  # invoke any scripts, etc from within the repo. I believe the fact that we'll
+  # be able to review PRs before this runs makes this fairly safe.
+  # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+  pull_request_target:
     types: [closed]
+  issue_comment:
+    types: [created]
```

Comment:
I know we are checking below to make sure the issue is a PR and has `/backport`, but will this cause this action to fire on _every_ issue comment and at least run that check? If so, are we concerned about that at all or should it not be an issue?

Reply:
Right, the action will fire on every issue comment and then immediately skip the comment. This is how the `/points` "Comment Commands" action works today as well, which you can see on some of our existing repos, like `up`:

<img width="938" alt="example" src="https://user-images.githubusercontent.com/1049349/120568694-3631ac80-c3c9-11eb-8533-40313d6b7440.png">

The two issues I could see here are:

1. Spamming our GitHub Actions log. I'm not too worried about this because you can select a specific workflow to filter out the noisy ones like Comment Commands.
2. Being billed for these Actions in our private repos. GitHub charges for actions by the minute, and doesn't provide any guidance on whether they bill for skipped actions. Skipped actions seem to take 3-5 seconds each to run, but the hover tooltip on the runtime explicitly says that time is not necessarily the same as what you'll be billed for.

One potential optimisation would be to merge the comment-handling variant into the "Comment Commands" flow such that we'd only be triggering one workflow per comment. The downside would be we'd be splitting the backport logic into two files.

Reply:
This sounds like something that, as long as we have good cost controls in place, we can enable and then see if it becomes a problem IMO

Reply:
I think I like the idea of moving the `/backport` into the Comment Commands action just to avoid two workflows triggering each time, so I'm going to do that.

Reply:
Updated - PTAL. Note that I'm inclined to avoid re-testing and hope that I got the Github syntax right, just because testing these actions is laborious. :|